File: 프로그래머스/unrated/132267. 콜라 문제/콜라 문제.js
function solution (need, refill, have) {
    let cola = 0;
    while(1) {
        if(have < need) break;
        if(have % need === 0) {
            have = (have/need)*refill;
            cola += have;
        } else {
            const keep = have % need;
            have = Math.floor(have/need)*refill;
            cola += have;
            have += keep;
        }
    }
    return cola;
}

File: 프로그래머스/1/132267. 콜라 문제/콜라 문제.py
import math

def solution(a, b, n):
    temp = n
    rest = 0
    count = 0
    
    while(True):
        if temp < a:
            return count
        
        cur_count = math.floor(temp / a) * b
        count += cur_count
        
        rest = temp % a
        temp = cur_count + rest
        
        
            
        
        
